libbeat/filter/condition.go

```diff
@@ -0,0 +1,330 @@
+	Range    map[string]RangeValue
+}
+
+func NewCondition(config ConditionConfig) (*Condition, error) {
+
+	c := Condition{}
+	c.Equals = map[string]EqualsValue{}
+	c.Contains = map[string]string{}
+	c.Regexp = map[string]*regexp.Regexp{}
+	c.Range = map[string]RangeValue{}
+
+	if err := c.AddEquals(config.Equals); err != nil {
+		return nil, err
+	}
+	if err := c.AddContains(config.Contains); err != nil {
+		return nil, err
+	}
+	if err := c.AddRegexp(config.Regexp); err != nil {
+		return nil, err
+	}
+	if err := c.AddRange(config.Range); err != nil {
+		return nil, err
+	}
+
+	return &c, nil
+}
+
+func (c *Condition) AddEquals(equals map[string]string) error {
+
+	for field, value := range equals {
+
+		i, err := strconv.Atoi(value)
+		if err == nil {
+			c.Equals[field] = EqualsValue{Int: i}
+		} else {
+			c.Equals[field] = EqualsValue{Str: value}
+		}
+	}
+	return nil
+}
+
+func (c *Condition) AddContains(contains map[string]string) error {
+
+	c.Contains = contains
+	return nil
+}
+
+func (c *Condition) AddRegexp(r map[string]string) error {
+
+	for field, value := range r {
+		reg, err := regexp.CompilePOSIX(value)
+		if err != nil {
+			return err
+		}
+		c.Regexp[field] = reg
+	}
+	return nil
+}
+
+func (c *Condition) AddRange(config map[string]RangeValue) error {
+
+	for field, rangeValue := range config {
+		c.Range[field] = rangeValue
+	}
+	return nil
+}
+
+func (c *Condition) Check(event common.MapStr) bool {
+
+	if !c.CheckEquals(event) {
+		return false
+	}
+	if !c.CheckContains(event) {
+		return false
+	}
+	if !c.CheckRegexp(event) {
+		return false
+	}
+	if !c.CheckRange(event) {
+		return false
+	}
+
+	return true
+}
+
+func (c *Condition) CheckEquals(event common.MapStr) bool {
+
+	for field, equalValue := range c.Equals {
+
+		value, err := event.GetValue(field)
+		if err != nil {
+			logp.Err("unavailable field %s: %v", field, err)
+			return false
+		}
+
+		switch value.(type) {
+		case uint8, uint16, uint32, uint64, int8, int16, int32, int64, int, uint:
+			return value == equalValue.Int
+		case string:
+			return value == equalValue.Str
+		default:
+			logp.Err("unexpected type %T in equals condition as it accepts only integers and strings. ", value)
+			return false
+		}
+
+	}
+
+	return true
+
+}
+
+func (c *Condition) CheckContains(event common.MapStr) bool {
+
+	for field, equalValue := range c.Contains {
+
+		value, err := event.GetValue(field)
+		if err != nil {
+			logp.Err("unavailable field %s: %v", field, err)
+			return false
+		}
+
+		switch value.(type) {
+		case string:
+			return strings.Contains(value.(string), equalValue)
+		default:
+			logp.Err("unexpected type %T in contains condition as it accepts only strings. ", value)
+			return false
+		}
+
+	}
+
+	return true
+
+}
+
+func (c *Condition) CheckRegexp(event common.MapStr) bool {
+
+	for field, equalValue := range c.Regexp {
+
+		value, err := event.GetValue(field)
+		if err != nil {
+			logp.Err("unavailable field %s: %v", field, err)
```

Comment:
Should we perhaps make this an info or debug message as it could lead to a lot of errors in case a field is filtered which does not exist in all events? Same above for CheckContains.